cloud-azure/src/main/java/com/sequenceiq/cloudbreak/cloud/azure/client/AzureClient.java

```diff
@@ -1006,6 +1007,19 @@ public void grantKeyVaultAccessPolicyToServicePrincipal(String resourceGroupName
         });
     }
 
+    public boolean checkKeyVaultAccessPolicyForServicePrincipal(String resourceGroupName, String vaultName, String principalObjectId) {
+        return handleAuthException(() -> {
+            List<AccessPolicy> accessPolicies = azure.vaults()
+                    .getByResourceGroup(resourceGroupName, vaultName)
+                    .accessPolicies();
+            if (accessPolicies != null) {
+                return accessPolicies.stream().anyMatch(policy -> principalObjectId.equals(policy.inner().objectId()));
```

Comment:
shouldn't we check these permissions? List.of(KeyPermissions.WRAP_KEY, KeyPermissions.UNWRAP_KEY, KeyPermissions.GET)

Reply:
@juhi-09 can you check Peter's comment?

Reply:
Added the check to validate the key permissions.